### PR TITLE
Remove DJWActionSheet

### PR DESCRIPTION
**Pull request by jschneier**
### Contributor checklist

- [x] I'm following the [code, UI and style conventions](https://github.com/WhisperSystems/Signal-iOS/blob/master/CONTRIBUTING.md#code-conventions)
- [x] My commits are rebased on the latest master branch
- [x] My commits are in nice logical chunks
- [x] My contribution is fully baked and is ready to be merged as is
- [x] I have tested my contribution on these devices:
 * iPhone SE iOS 10.1.1
- [x] I have made the choice whether I want the [BitHub reward](https://github.com/WhisperSystems/Signal-Android/wiki/BitHub-Rewards) or not by omitting or adding the word `FREEBIE` in my commit message

- - - - - - - - - -

### Description

Supercedes #1106. I mostly followed your lead but had to make some decisions where things have diverged. I have a couple of outstanding questions such as if we are okay with the balance of including/not including titles & messages that I have in here or not. I'm also not sure how to generate the corrupted session (clearing it I made an alert) or to get the message failed to send popup. The other ones look good to me.

Also, for some reason I currently have the following thing happening but I'm not sure if it's related to this or if it's an issue in SignalServiceKit. (I intentionally unregistered and reregistered the device so I could get to the action sheet).

screen shot 2016-11-26 at 7 34 54 pm

### CODE REVIEW

File: Signal/src/view controllers/NewGroupViewController.m [hunk 1]
```diff
@@ -238,34 +237,29 @@ - (TSGroupModel *)makeGroup
...
-                                    break;
-                            }
-                        }
-                      }];
+    UIAlertController *actionSheetController = [UIAlertController alertControllerWithTitle:NSLocalizedString(@"NEW_GROUP_ADD_PHOTO_ACTION", @"Action Sheet title prompting the user for a group avatar")
+                                                                                   message:nil
+                                                                            preferredStyle:UIAlertControllerStyleActionSheet];
+    UIAlertAction *dismissAction = [UIAlertAction actionWithTitle:NSLocalizedString(@"TXT_CANCEL_TITLE", @"")
+                                                            style:UIAlertActionStyleCancel
+                                                          handler:^(UIAlertAction * _Nonnull action) { /*no-op*/ } ];
```
Comment (michaelkirk) [inline Signal/src/view controllers/NewGroupViewController.m]: replace w/ nil

**Comment by michaelkirk** [comment]
Also, the "ERROR_MESSAGE_WRONG_TRUSTED_IDENTITY_KEY" is explained by the outdated strings file. It's missing the localization for that string.

**Comment by jschneier** [comment]
@michaelkirk the translations thing is because I need to have a cloned copy of SignalServiceKit in the parent directory, that is non-obvious. I have fixed that locally but now the missing `EPContactsPicker` worries me.

This is what I get from running the binary now:

```
 ~/c/Signal-iOS (remove-djw)> Signal/translations/bin/auto-genstrings
find: Libraries/EPContactsPicker: No such file or directory
Bad entry in file Pods/JSQMessagesViewController/JSQMessagesViewController/Categories/NSBundle+JSQMessages.m (line = 39): Argument is not a literal string.
```

Updated everything else as requested.

**Comment by michaelkirk** [comment]
Thanks!

**Comment by michaelkirk** [comment]
I've fixed up the errors you were experiencing with the `auto-genstrings` script here: https://github.com/WhisperSystems/Signal-iOS/pull/1420. Thanks for bringing it up.

**Comment by jschneier** [comment]
Looking good!

**Comment by michaelkirk** [comment]
TBH I’m not really sure. I just know it was not finished when I arrived on the scene, but bits of lingering code referencing it remain.

I’ve detailed my understanding of what needs to happen here:
https://github.com/WhisperSystems/Signal-iOS/issues/230 <https://github.com/WhisperSystems/Signal-iOS/issues/230>

**Pull request merged**